v2/src/datadog.ts

```diff
@@ -185,3 +189,10 @@ function grantReadLambdas(secret: ISecret, lambdaFunctions: lambda.Function[]) {
     secret.grantRead(functionName);
   });
 }
+
+function grantReadLambdasFromSecretArn(construct: Construct, arn: string, lambdaFunctions: lambda.Function[]) {
+  const secret = Secret.fromSecretPartialArn(construct, "DatadogApiKeySecret", arn);
+  lambdaFunctions.forEach((functionName) => {
+    secret.grantRead(functionName);
```

Comment:
Users deploying these applications may not have permissions to grant read access to keys in SSM or KMS - is there a way we can make this optional?

Reply:
if users don't pass the construct to `addLambdaFunctions` we won't try to grant anything. We could document this explicitly, only pass in the construct if you want us to try auto-granting secret read permissions?

Reply:
the optional construct param is only used for this feature

Reply:
We could, but we may need the construct for other work in the near future.

Can we default this to true but allow users to disable this behavior if needed?

It's (occasionally) common to see devops teams own secrets management and have elevated permissions to create or grant read permissions to individual applications, so that individual developer teams can't read arbitrary keys out of secrets storage.

Reply:
Gotcha, makes sense. I'll add a config param for disabling it!